src/openeo_aggregator/backend.py

```diff
@@ -672,14 +677,60 @@ def _get_connection_and_backend_service_id(
 
     def service_types(self) -> dict:
         """https://openeo.org/documentation/1.0/developers/api/reference.html#operation/list-service-types"""
-        # TODO: add caching. Also see https://github.com/Open-EO/openeo-aggregator/issues/78#issuecomment-1326180557
+        cached_info = self._memoizer.get_or_call(key=("_get_service_types",), callback=self._get_service_types)
+        # Convert the cached results back to the format that service_types should return.
+        return {name: data["service_type"] for name, data, in cached_info.items()}
+
+    def _find_backend_id_for_service_type(self, service_type: str) -> Optional[str]:
+        """Returns the ID of the backend that provides the service_type or None if no backend supports that service."""
+        cached_info = self._memoizer.get_or_call(key=("_get_service_types",), callback=self._get_service_types)
```

Comment:
instead of having duplicate `self._memoizer.get_or_call(key=("_get_service_types",), callback=self._get_service_types)`  calls , I would encapsulate that in a `_get_service_types_cached` .

see `AggregatorCollectionCatalog._get_all_metadata_cached`

Reply:
Implemented.